specs/core/0_beacon-chain.md

```diff
@@ -2310,6 +2341,60 @@ def process_attester_slashing(state: BeaconState,
         slash_validator(state, index)
 ```
 
+##### Randao reveal slashings
+
+Verify that `len(block.body.randao_reveal_slashings) <= MAX_RANDAO_REVEAL_SLASHINGS`.
+
+For each `randao_reveal_slashing` in `block.body.randao_reveal_slashings`, run the following function:
+
+```python
+def process_randao_reveal_slashing(state: BeaconState,
+                                   randao_reveal_slashing: RandaoRevealSlashing) -> None:
+    """
+    Process ``RandaoRevealSlashing`` operation.
+    Note that this function mutates ``state``.
+    """
+    revealer = state.validator_registry[randao_reveal_slashing.revealer_index]
+    masker = state.validator_registry[randao_reveal_slashing.masker_index]
+    pubkeys = [revealer.pubkey, masker.pubkey]
+    message_hashes = [
+        hash_tree_root(randao_reveal_slashing.epoch),
+        randao_reveal_slashing.mask,
+    ]
+
+    assert randao_reveal_slashing.epoch >= get_current_epoch(state) + RANDAO_SLASHING_EPOCHS
+    assert revealer.slashed is False
+    assert randao_reveal_slashing.epoch not in state.validator_registry[randao_reveal_slashing.revealer_index].exposed_randao_reveals
+
+    assert bls_verify_multiple(
+        pubkeys=pubkeys,
+        message_hashes=message_hashes,
+        signature=randao_reveal_slashing.reveal,
+        domain=get_domain(
+            fork=state.fork,
+            epoch=randao_reveal_slashing.epoch,
+            domain_type=DOMAIN_RANDAO,
+        ),
+    )
+
+    if randao_reveal_slashing.epoch >= get_current_epoch(state) + CUSTODY_PERIOD_TO_RANDAO_PADDING:
+        # Replacement for custody reveal slashing
+        slash_validator(state, randao_reveal_slashing.revealer_index, randao_reveal_slashing.masker_index)
+    else:
+        # Only a small penalty for RANDAO reveal that does not interfere with the custody period
+        penalty = get_effective_balance(state, randao_reveal_slashing.revealer_index) // RANDAO_REVEAL_PENALTY_QUOTIENT
```

Comment:
```suggestion
        max_proposer_slot_reward = (
            get_base_reward(state, randao_reveal_slashing.revealer_index) *
            len(get_active_validators(state, get_current_epoch(state))) //
            EPOCH_LENGTH  //
            PROPOSER_REWARD_QUOTIENT
        )
        penalty = max_proposer_slot_reward * 2
```

Reply:
Propose to make the penalty be proportional to the reward a proposer can receive in a slot.

Reply:
OK, I assume you meant
```
        max_proposer_slot_reward = (
            get_base_reward(state, randao_key_reveal.revealed_index) *
            SLOTS_PER_EPOCH //
            len(get_active_validator_indices(state, get_current_epoch(state))) //
            PROPOSER_REWARD_QUOTIENT
        )
```
(the quotient is inverted)